rpcs3/rpcs3qt/gui_application.cpp

```diff
@@ -1180,23 +1197,26 @@ void gui_application::OnAppStateChanged(Qt::ApplicationState state)
 bool gui_application::native_event_filter::nativeEventFilter([[maybe_unused]] const QByteArray& eventType, [[maybe_unused]] void* message, [[maybe_unused]] qintptr* result)
 {
 #ifdef _WIN32
-	if (!Emu.IsRunning() && !g_raw_mouse_handler)
+	if (!Emu.IsRunning() && !Emu.IsStarting())
 	{
 		return false;
 	}
 
 	if (eventType == "windows_generic_MSG")
 	{
-		if (MSG* msg = static_cast<MSG*>(message); msg && (msg->message == WM_INPUT || msg->message == WM_KEYDOWN || msg->message == WM_KEYUP))
+		if (MSG* msg = static_cast<MSG*>(message); msg)
 		{
-			if (auto* handler = g_fxo->try_get<MouseHandlerBase>(); handler && handler->type == mouse_handler::raw)
-			{
-				static_cast<raw_mouse_handler*>(handler)->handle_native_event(*msg);
+			if (g_raw_mouse_handler && (msg->message == WM_INPUT || msg->message == WM_KEYDOWN || msg->message == WM_KEYUP)) {
+				if (auto* handler = g_fxo->try_get<MouseHandlerBase>(); handler && handler->type == mouse_handler::raw)
+				{
+					static_cast<raw_mouse_handler*>(handler)->handle_native_event(*msg);
+				}
+				g_raw_mouse_handler->handle_native_event(*msg);
 			}
-
-			if (g_raw_mouse_handler)
+			if (msg->message == WM_DEVICECHANGE)
 			{
-				g_raw_mouse_handler->handle_native_event(*msg);
+				handle_hotplug_event(msg->wParam == DBT_DEVICEARRIVAL);
+				return false;
```

Comment:
This is all completely wrong. See my last review suggestion for this file

Reply:
Sorry yeah i now understand what you were suggesting, fixed.